gnomad_qc/v4/sample_qc/relatedness.py

```diff
@@ -0,0 +1,251 @@
+            # `collect`.
+            logger.info("Writing metadata...")
+            col_idx_mapping = hl.struct(col_idx=mt.col_idx, s=mt.s).collect()
+            col_idx_mapping.sort(key=lambda e: e.col_idx)
+            metadata = {
+                "num_sites": row_count_v4,
+                "samples": [e.s for e in col_idx_mapping],
+            }
+            with hadoop_open(f"{input_path}/metadata.json", 'w') as f:
+                json.dump(metadata, f)
+
+        if args.create_relatedness_table:
+            from hail.utils.java import Env
+
+            spark = Env.spark_session()
+            df = spark.read.parquet(cuking_output_path(test=test))
+            ht = hl.Table.from_spark(df)
+            ht = ht.repartition(64)
+            ht = ht.key_by(ht.i, ht.j)
+            ht.write(relatedness.path)
+
+        if args.compute_related_samples_to_drop:
+            # compute_related_samples_to_drop uses a rank table as a tie breaker when
+            # pruning samples. We determine rankings as follows: filtered samples are
+            # ranked lowest, followed by non-releasable samples, followed by presence in
+            # v3 (favoring v3 over v4), and then by coverage.
+            def annotate_project_meta_ht(
+                project_meta_ht, sex_ht, hard_filtered_samples_ht
+            ):
+                return project_meta_ht.select(
+                    project_meta_ht.releasable,
+                    project_meta_ht.exclude,
+                    chr20_mean_dp=sex_ht[project_meta_ht.key].chr20_mean_dp,
+                    filtered=hl.or_else(
+                        hl.len(
+                            hard_filtered_samples_ht[project_meta_ht.key].hard_filters
+                        )
+                        > 0,
+                        False,
+                    ),
+                )
+
+            from gnomad_qc.v3.resources.meta import project_meta as project_meta_v3
+            from gnomad_qc.v4.resources.meta import project_meta as project_meta_v4
+            from gnomad_qc.v3.resources.sample_qc import (
+                hard_filtered_samples as hard_filtered_samples_v3,
+            )
+            from gnomad_qc.v4.resources.sample_qc import (
+                hard_filtered_samples as hard_filtered_samples_v4,
+            )
+            from gnomad_qc.v3.resources.sample_qc import sex as sex_v3
+            from gnomad_qc.v4.resources.sample_qc import sex as sex_v4
+
+            annotated_project_meta_v3_ht = annotate_project_meta_ht(
+                project_meta_v3.ht(), sex_v3.ht(), hard_filtered_samples_v3.ht()
+            )
+            annotated_project_meta_v4_ht = annotate_project_meta_ht(
+                project_meta_v4.ht(), sex_v4.ht, hard_filtered_samples_v4.ht()
+            )
+
+            rank_ht = annotated_project_meta_v3_ht.union(
+                annotated_project_meta_v4_ht, unify=True
+            )
+
+            rank_ht = rank_ht.order_by(
```

Comment:
Need to add that we want to favor v3 `release` samples, then `v4 samples` over v3 non release samples

Reply:
Thanks for catching that! I'll fix that once you've confirmed whether we could potentially merge the metadata.

Reply:
Added PR for the metadata merge: https://github.com/broadinstitute/gnomad_qc/pulls